snuba/cli/migrations.py

```diff
@@ -123,3 +127,68 @@ def reverse(
         raise click.ClickException(str(e))
 
     click.echo(f"Finished reversing migration {migration_key}")
+
+
+@migrations.command()
+@click.option(
+    "--type", "node_type", type=click.Choice(["local", "dist"]), required=True
+)
+@click.option(
+    "--storage-set",
+    "storage_set_names",
+    type=click.Choice([s.value for s in StorageSetKey]),
+    required=True,
+    multiple=True,
+)
+@click.option("--host-name", type=str, required=True)
+@click.option("--port", type=int, required=True)
+@click.option("--user", type=str, required=True, default="default")
+@click.option("--password", type=str, required=True, default="")
+@click.option("--database", type=str, required=True, default="default")
+def add_node(
+    node_type: str,
+    storage_set_names: Sequence[str],
+    host_name: str,
+    port: int,
+    user: str,
+    password: str,
+    database: str,
+) -> None:
+    """
+    Runs all migrations on a brand new ClickHouse node. This should be performed
+    before a new node is added to an existing ClickHouse cluster.
+
+    All of the SQL operations for the provided storage sets will be run. Any non
+    SQL (Python) operations will be skipped.
+
+    This operation does not change the migration status in the migrations_local
+    / migrations_dist tables, since it is designed to bring a new node up to
+    the same state as existing ones already added to the cluster.
```

Comment:
How does this fit in the plan for supporting distributed migrations? 
If we use this to bring a new node up to speed. How do we plan to keep it up to date following this migration? The runner would still fail if we try to apply migrations to nodes in a distributed deployment.

Reply:
This command exists to bring a node up to the same state as all the other nodes, allowing it to be added to the ClickHouse cluster. You can't add a node to the cluster before it has the required tables on it, which means it cannot be discovered by the migration system yet. Once it is added to the cluster it should be managed in the same way as the other nodes. Eventually this will be with the runner but yes, for now it is manual.

Reply:
ok